iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/IcebergTableUtil.java

```diff
@@ -52,12 +52,13 @@ private IcebergTableUtil() {
    * Constructs the table properties needed for the Iceberg table loading by retrieving the information from the
    * hmsTable. It then calls {@link IcebergTableUtil#getTable(Configuration, Properties)} with these properties.
    * @param configuration a Hadoop configuration
-   * @param hmsTable the HMS table
-   * @param skipCache if set to true there won't be an attempt to retrieve the table from SessionState
+   * @param hmsTable      the HMS table
+   * @param skipCache     if set to true there won't be an attempt to retrieve the table from SessionState
+   * @param suffix        the suffix to use for cache.
    * @return the Iceberg table
    */
   static Table getTable(Configuration configuration, org.apache.hadoop.hive.metastore.api.Table hmsTable,
-      boolean skipCache) {
+      boolean skipCache, String suffix) {
```

Comment:
Why do we need the suffix?

Reply:
We want to cache the table for Stats only, if you fetch once for Stats, then use that till end, if you fetch for something else, we don't use that.

In most cases that won't be a problem, but the way we do CTAS and MV's it did creates problem. It first creates an iceberg table and then does the load into that, so in that cases caching would have stale table. Got a test failure as well for those MV cases in the last run

Reply:
what do you mean only for stats, we don't have an unlimited heap. if you know that stats is stale in certain cases (CTAS, MV) - invalidate the cache.

Reply:
I changed the approach, if it is write operation where the rows affected gets changed, I read from the Cache, if not do the same what we do today.

Give a check, if that sounds cool, or if you feel if it that can also cause some issues,